### PR TITLE
Post handshake KeyUpdate support

### LINKED ISSUE #210: Post handshake KeyUpdate support
Add support for updating keying material after handshake

### PR DESCRIPTION
**Pull request by NoName115**
Support for handling KeyUpdate post-handshake msg.

fixes #210

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlslite-ng/341)

### CODE REVIEW

**Comment by tomato42** [comment]
This pull request **introduces 2 alerts** and **fixes 7** when merging b84a30e6f3668da0b0345c6027433ddb20141cd4 into 029425144f6b26d5ab8acc3a79e3ead79253ad71 - [view on LGTM.com](https://lgtm.com/projects/g/tomato42/tlslite-ng/rev/pr-191778fbffaedeca392d7ef240dca4c50d13f76d)

**new alerts:**

* 1 for \_\_eq\_\_ not overridden when adding attributes
* 1 for Syntax error

**fixed alerts:**

* 6 for &#39;import \*&#39; may pollute namespace
* 1 for Testing equality to None

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by NoName115** [comment]
Updated.

**Comment by tomato42** [comment]
This pull request **introduces 1 alert** when merging 8cd0e174fe2bfcdc6d413804d3bbb4f3c9f4c1cd into 029425144f6b26d5ab8acc3a79e3ead79253ad71 - [view on LGTM.com](https://lgtm.com/projects/g/tomato42/tlslite-ng/rev/pr-87f1c313a0f3f03735976f8506ec67581ffe326d)

**new alerts:**

* 1 for \_\_eq\_\_ not overridden when adding attributes

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by NoName115** [comment]
Updated.

**Comment by NoName115** [comment]
Fixed.

**Comment by tomato42** [comment]
This pull request **introduces 1 alert** when merging 7f889c254f3f94d59dc3bc6da1a52aeada28716e into 029425144f6b26d5ab8acc3a79e3ead79253ad71 - [view on LGTM.com](https://lgtm.com/projects/g/tomato42/tlslite-ng/rev/pr-e1668a90fee8aeccbee8abc7aba852a4fd4f018d)

**new alerts:**

* 1 for \_\_eq\_\_ not overridden when adding attributes

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by tomato42** [comment]
This pull request **introduces 1 alert** when merging d8100439b80e569c9254af312e07ea71c09c6478 into 029425144f6b26d5ab8acc3a79e3ead79253ad71 - [view on LGTM.com](https://lgtm.com/projects/g/tomato42/tlslite-ng/rev/pr-fa38921ec806e9598b2d9750bd7e850aad73507f)

**new alerts:**

* 1 for \_\_eq\_\_ not overridden when adding attributes

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by tomato42** [comment]
This pull request **introduces 1 alert** when merging 637c5c6f5eee3bace0893955411458323e6c140d into 029425144f6b26d5ab8acc3a79e3ead79253ad71 - [view on LGTM.com](https://lgtm.com/projects/g/tomato42/tlslite-ng/rev/pr-48aa3a491a6c6db48d3c7c7b322b0e5e329a73e4)

**new alerts:**

* 1 for \_\_eq\_\_ not overridden when adding attributes

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: tlslite/tlsrecordlayer.py [hunk 1]
```diff
@@ -315,6 +316,10 @@ def readAsync(self, max=None, min=1):
                         result.time = time.time()
                         self.tickets.append(result)
                         continue
+                    if isinstance(result, KeyUpdate):
+                        for _ in self._handle_keyupdate_request(result):
+                            pass
```
Comment (tomato42) [inline tlslite/tlsrecordlayer.py]: it should actually yield result here too - when the writing is blocking it will hang here

**Comment by NoName115** [comment]
updated.

**Comment by tomato42** [comment]
This pull request **introduces 2 alerts** when merging 3334691bd26b6338b5dfc5acab6b624cf921a196 into 029425144f6b26d5ab8acc3a79e3ead79253ad71 - [view on LGTM.com](https://lgtm.com/projects/g/tomato42/tlslite-ng/rev/pr-905a948ada0055c489d39d479cd5b3f99d6d9d75)

**new alerts:**

* 1 for \_\_eq\_\_ not overridden when adding attributes
* 1 for Unused local variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by tomato42** [comment]
This pull request **introduces 1 alert** when merging 64761b131e3e45dba55f43460a47792663087066 into 029425144f6b26d5ab8acc3a79e3ead79253ad71 - [view on LGTM.com](https://lgtm.com/projects/g/tomato42/tlslite-ng/rev/pr-3d1e086ac2a12b01eafd7773e304fcc13cc57140)

**new alerts:**

* 1 for \_\_eq\_\_ not overridden when adding attributes

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by tomato42** [comment]
This pull request **introduces 1 alert** when merging 22ae125b770d11b8840aad04874e2c78c7fe1644 into 849f1d15cd49ec7fc62a4c5564f32fd857791ba7 - [view on LGTM.com](https://lgtm.com/projects/g/tomato42/tlslite-ng/rev/pr-2b9b6ac93a35d52efedb44ccceea1667069db999)

**new alerts:**

* 1 for \_\_eq\_\_ not overridden when adding attributes

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by tomato42** [comment]
This pull request **introduces 1 alert** when merging d3a4b823538aa5492d611130b343056c49cacadd into 849f1d15cd49ec7fc62a4c5564f32fd857791ba7 - [view on LGTM.com](https://lgtm.com/projects/g/tomato42/tlslite-ng/rev/pr-c34e3d1883ca4cda9a4a003ead1a1af6dd77ff36)

**new alerts:**

* 1 for \_\_eq\_\_ not overridden when adding attributes

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by tomato42** [comment]
The issue returned by LGTM is a false positive: https://github.com/Semmle/ql/issues/2408

**Comment by tomato42** [comment]
remaining test coverage is provided by https://github.com/tomato42/tlsfuzzer/pull/501

**Comment by The-Mule** [comment]
I checked test coverage only. The rest was already reviewed. AFAIK all changes requested were already addressed and this can be merged, finally.

**Pull request merged**